internal/transport/http2_client.go

```diff
@@ -244,6 +244,9 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 		<-newClientCtx.Done()       // Block until connectCtx expires or the defer above executes.
 		if connectCtx.Err() != nil {
 			// connectCtx expired before exiting the function.  Hard close the connection.
+			if logger.V(logLevel) {
+				logger.Infof("transport: closing connection due to connect context expiring.")
```

Comment:
`s/transport/newClientTransport/`?  I think it's important to distinguish that we don't even have a transport at this point.